### DISCUSSION TITLE
Copyright headers in project templates

### DISCUSSION THREAD

**Discussion by karlssten** [eclipse-glsp/glsp#1220] [General] [answered]
What is the purpose of having copyright headers in the templates provided in the glsp-examples repository? I would like to use one as a starting point for my project, but I don't want to have the copyright headers in it. How are the templates meant to be used? Should they be called templates if they are only meant to be used for reference and not cloned and adapted?

Thank you!

**Comment by planger** [comment, marked as answer]
Thanks for reaching out with your question! The presence of copyright headers in the templates from the glsp-examples repository is indeed an obligatory practice for projects under the Eclipse Foundation (see [project handbook](https://www.eclipse.org/projects/handbook/#ip-copyright-headers)). These headers are there to provide clarity about the ownership and licensing of the code, ensuring that users are aware of their rights and obligations when using it.

That being said, we totally get your point about wanting to use these templates as a starting point for your own projects. And that's exactly what they're there for! To make this as easy as possible, the templates are dual-licensed, which includes the permissive [MIT License](https://github.com/eclipse-glsp/glsp-examples/blob/master/LICENSE-MIT). This means you're free to clone, adapt, and use the templates in your projects without the need to publish your modifications.

**Reply by karlssten** [reply]
Thank you for your answer. One last thing, does that mean that the copyright headers & licence have to be left in the adapted code?

**Reply by planger** [reply]
Yes, afaik, copyright headers always should remain in the file. You're free to add yourself or your company if you enhance the source code though.
